LeanCode.CodeAnalysis.ruleset

```diff
@@ -9,6 +9,8 @@
         <!-- ConvertSwitchStatementToExpression -->
         <Rule Id="IDE1006" Action="None"/>
         <!-- NamingRule (interface with I) - we already have this rule in StyleCop -->
+        <Rule Id="CA1816" Action="None"/>
+        <!-- DisposeMethodsShouldCallSuppressFinalize - we don't use finalizers that much -->
```

Comment:
Do we actually violate this rule so much?

Reply:
Yeah, it fires up even if you don't have finalizer.

Reply:
Then maybe consider either suppressing it on each `Dispose` method we have or just call that `GC.SuppressFinalize(this)` to satisty it, even if it's a noop in most cases. You never know when the time for implementing a finalizer actually comes, and then such a rule might be useful.

Reply:
Up to the point when we really need the finalizer, this rule just clutters the code. I think we never used the finalizer in our code and I don't see it being used in the near future.